lib/srv/alpnproxy/common/protocols.go

```diff
@@ -129,9 +138,56 @@ func ToALPNProtocol(dbProtocol string) (Protocol, error) {
 // to terminated DB connection.
 func IsDBTLSProtocol(protocol Protocol) bool {
 	switch protocol {
-	case ProtocolMongoDB, ProtocolRedisDB, ProtocolSQLServer, ProtocolSnowflake:
+	case ProtocolMongoDB, ProtocolWithPing(ProtocolMongoDB),
+		ProtocolRedisDB, ProtocolWithPing(ProtocolRedisDB),
+		ProtocolSQLServer, ProtocolWithPing(ProtocolSQLServer),
+		ProtocolSnowflake, ProtocolWithPing(ProtocolSnowflake):
 		return true
 	default:
 		return false
 	}
 }
+
+// ProtocolsWithPingSupport is the list of protocols that Ping connection is
+// supported.
+var ProtocolsWithPingSupport = []Protocol{
+	ProtocolPostgres,
+	ProtocolMySQL,
+	ProtocolMongoDB,
+	ProtocolRedisDB,
+	ProtocolSQLServer,
+	ProtocolSnowflake,
+}
```

Comment:
I kept it in a separate definition since I wasn't sure if we wanted to add other protocols (for example, application access) to have ping support. If this is a database thing, I agree we could only define it once.